src/Mod/Sketcher/Gui/DrawSketchHandlerRotate.h

```diff
@@ -0,0 +1,627 @@
+                   QT_TRANSLATE_NOOP(
+                       "Notifications",
+                       "Tool execution aborted") "\n")  // This prevents constraints from being
+                                                        // applied on non existing geometry
+        }
+    }
+
+    void createAutoConstraints() override
+    {
+        // none
+    }
+
+    std::string getToolName() const override
+    {
+        return "DSH_Rotate";
+    }
+
+    QString getCrosshairCursorSVGName() const override
+    {
+        return QString::fromLatin1("Sketcher_Pointer_Create_Rotate");
+    }
+
+    std::unique_ptr<QWidget> createWidget() const override
+    {
+        return std::make_unique<SketcherToolDefaultWidget>();
+    }
+
+    bool isWidgetVisible() const override
+    {
+        return true;
+    };
+
+    QPixmap getToolIcon() const override
+    {
+        return Gui::BitmapFactory().pixmap("Sketcher_Rotate");
+    }
+
+    QString getToolWidgetText() const override
+    {
+        return QString(QObject::tr("Rotate parameters"));
+    }
+
+    void activated() override
+    {
+        DrawSketchDefaultHandler::activated();
+        continuousMode = false;
+    }
+
+private:
+    std::vector<int> listOfGeoIds;
+    Base::Vector2d centerPoint, startPoint, endpoint;
+
+    bool deleteOriginal, cloneConstraints;
+    double length, startAngle, endAngle, totalAngle, individualAngle;
+    int numberOfCopies;
+
+    void deleteOriginalGeos()
+    {
+        std::stringstream stream;
+        for (size_t j = 0; j < listOfGeoIds.size() - 1; j++) {
+            stream << listOfGeoIds[j] << ",";
+        }
+        stream << listOfGeoIds[listOfGeoIds.size() - 1];
+        try {
+            Gui::cmdAppObjectArgs(sketchgui->getObject(),
+                                  "delGeometries([%s])",
+                                  stream.str().c_str());
+        }
+        catch (const Base::Exception& e) {
+            Base::Console().Error("%s\n", e.what());
+        }
+    }
+
+    void createShape(bool onlyeditoutline) override
+    {
+        Sketcher::SketchObject* Obj = sketchgui->getSketchObject();
+
+        ShapeGeometry.clear();
+
+        int numberOfCopiesToMake = numberOfCopies;
+        if (numberOfCopies == 0) {
+            numberOfCopiesToMake = 1;
+            deleteOriginal = true;
+        }
+        else {
+            deleteOriginal = false;
+        }
+
+        individualAngle = totalAngle / numberOfCopiesToMake;
+
+        for (int i = 1; i <= numberOfCopiesToMake; i++) {
+            for (auto& geoId : listOfGeoIds) {
+                auto geo = std::unique_ptr<Part::Geometry>(Obj->getGeometry(geoId)->copy());
+
+                bool isConstructionGeo =
+                    Sketcher::GeometryFacade::getConstruction(Obj->getGeometry(geoId));
+                Sketcher::GeometryFacade::setConstruction(geo.get(), isConstructionGeo);
+                double angle = individualAngle * i;
+
+                if (isCircle(*geo)) {

```

Comment:
There is nothing wrong in using this.

However, now we have the is template function, and I think we should move to use that more generic:
```geo->is<Part::GeomCircle>();```

You do not need to change it, it is more of an indication. I will migrate all calls eventually as a separate PR.

Reply:
imo the syntax isCircle(*geo) is cleaner and shorter.
Best would be `geo->isCircle();`